app/main/helpers/__init__.py

```diff
@@ -8,3 +11,13 @@ def hash_email(email):
     m.update(email.encode('utf-8'))
 
     return base64.urlsafe_b64encode(m.digest())
+
+
+def supplier_role_required(func):
```

Comment:
This is true whenever we require a logged in user, I'd just wrap `@login_required` with additional role check, give it the same `login_required` name and then fix the imports – just to make sure we're never missing role checks.